src/mountscript.sh

```diff
@@ -602,6 +652,329 @@ ensure_aznfswatchdog()
+            return 1
+        fi
+
+        used_port=$(cat $stunnel_conf_file | grep accept | cut -d: -f2)
+        vecho "used port: '$used_port'"
+
+        chattr -f -i $stunnel_conf_file
+
+        sed -i "s/$used_port/$available_port/" $stunnel_conf_file
+        if [ $? -ne 0 ]; then
+            eecho "Failed to replace the port in $stunnel_conf_file."
+            chattr -f +i $stunnel_conf_file
+            return 1
+        fi
+        chattr -f +i $stunnel_conf_file
+
+        new_used_port=$(cat $stunnel_conf_file | grep accept | cut -d: -f2)
+
+        # start the stunnel process
+        stunnel_status=$(stunnel $stunnel_conf_file 2>&1)
+        if [ -n "$stunnel_status" ]; then
+            is_binding_error=$(echo $stunnel_status | grep "$LOCALHOST:$new_used_port: Address already in use")
+            if [ -z "$is_binding_error" ]; then
+                eecho "[FATAL] Not able to start stunnel process for '${stunnel_conf_file}'!"
+                return 1
+            fi
+        else
+	        vecho "Found new port '$new_used_port' and restarted stunnel."
+	        break
+        fi
+    done
+}
+
+get_next_available_port()
+{
+    for ((port=NFSV4_PORT_RANGE_START; port<=NFSV4_PORT_RANGE_END; port++))
+    do
+        is_port_available=`netstat -tuapn | grep "$LOCALHOST:$port "`
+        if [ -z "$is_port_available" ]; then
+            break
+        fi
+    done
+
+    if [ $port -le $NFSV4_PORT_RANGE_END ]; then
+        echo "$port"
+    else
+        echo ""
+    fi
+}
+
+
+install_CA_cert()
+{
+    wget https://cacerts.digicert.com/DigiCertGlobalRootG2.crt.pem --no-check-certificate -O /usr/local/share/ca-certificates/DigiCert_Global_Root_G2.crt
+    if [ $? -ne 0 ]; then
+        eecho "[FATAL] Not able to download DigiCert_Global_Root_G2 certificate from https://cacerts.digicert.com/DigiCertGlobalRootG2.crt.pem !"
+        return 1
+    fi
+
+    update-ca-certificates
+}
+
+
+#
+# Add stunnel configuration in stunnel_<storageaccount>.conf file.
+#
+add_stunnel_configuration()
+{
+    chattr -f -i $stunnel_conf_file
+
+
+    if [ ! -f $STUNNEL_CAFILE ]; then
+        vecho "CA root cert is missing for stunnel configuration. Installing DigiCert_Global_Root_G2 certificate."
+        install_CA_cert
+        if [ $? -ne 0 ]; then
+            chattr -f +i $stunnel_conf_file
+            eecho "[FATAL] Not able to install DigiCert_Global_Root_G2 certificate!"
+            return 1
+        fi
+    fi
+
+    echo "CAFile = $STUNNEL_CAFILE" >> $stunnel_conf_file
+    if [ $? -ne 0 ]; then
+        chattr -f +i $stunnel_conf_file
+        eecho "Failed to add CAFile path to $stunnel_conf_file!"
+        return 1
+    fi
+
+    echo "verifyChain = yes" >> $stunnel_conf_file
+    if [ $? -ne 0 ]; then
+        chattr -f +i $stunnel_conf_file
+        eecho "Failed to add verifyChain option to $stunnel_conf_file!"
+        return 1
+    fi
+
+    # TODO: checkHost value could be different for prod tenants.
```

Comment:
It will for sure different for prod tenants, and we cannot tell if the mount helper is called from preprod or prod. Also depending on how the certificate is created by certificate team, this value maybe different between preprod and prod. that's the reason I said we may need an extra flag to turn off this check, but we don't have to deal it right now here.
please do verify this changes works for devfabric.

Reply:
I replied to your comment from previous iteration and explained how we can switch the value of checkHost for preprod and prod.
Also, I verified that this checHost works for devfabric.